README.md

```diff
@@ -1,16 +1,68 @@
 # helm-migrate-values
 
-## Documentation
-
-- [Install](#install)
-
+A plugin to migrate user-specified Helm values between chart versions when the schema of `values.yaml` changes. Define migration paths with migration files in the chart repo to ensure seamless upgrades.
 ## Requirements
 
-- Helm version 3 or newer
+- **Helm** version 3 or newer
+- **Git** as the `helm plugin install` command requires it
 
 ## Install
 
 ```
 $ helm plugin install https://github.com/OctopusDeploy/helm-migrate-values.git
 ```
 
+## Usage
+
+This Helm plugin enables users to migrate values across chart versions, accounting for changes in the values.yaml schema.
+
+> **_NOTE:_** The intended usage of this plugin is that it does not apply any changes to the release on its own. It simply outputs the values to override, which are meant to be used with `helm upgrade --reset-then-reuse-values`.
+> 
+> See [output values](#optional-output-the-migration-to-a-file) section for an example on how to apply the migration to a release 
+
+### Step 1: Define Migration Files
+Start by defining migration files within your Helm chart. These files should be placed under the `value-migrations/` directory, relative to your base chart directory. You can customize the migration directory location by using the `--migration-dir` flag.
+
+#### Migration File Naming Convention
+Each migration file should follow the naming format:
+`to-v{VERSION_TO}.yaml`, where **VERSION_TO** is the target chart version. The plugin will use this file to define the transformation between the previous version and the specified version.
+
+#### Migration File Structure
+These migration files are written in YAML, using Go templating for transforming and mapping values from the old schema to the new one. See this [example](pkg/test-charts/v2/value-migrations/to-v2.yaml) from the integration test.
+
+### Step 2: Run the Migration
+To migrate your Helm release to a new chart version, use the following command:
+```
+helm migrate-values [RELEASE] [CHART] [flags]
+```
+
+- **RELEASE**: The name of the Helm release you're migrating.
+- **CHART**: The chart you're migrating to, which can be a local chart (specified by file path) or a remote chart (using the `oci://` prefix).
+
+## Example
+```
+helm migrate-values my-kubernetes-agent oci://registry-1.docker.io/octopusdeploy/kubernetes-agent \
+  --version 2.4.0 \
+  -n octopus-agent-demo \
+  --migration-dir kubernetes-agent/value-migrations
+```
+
+### Optional: Output the Migration to a File
```

Comment:
While this is _technically_ optional, the only real use case for not having it is for testing. I would maybe reverse this so that the example above has the file specified, then change this section to explain that you don't have to specify the file if you just want to output the migrated values to stdout.